src/lightmap.cpp

```diff
@@ -469,6 +470,171 @@ bool map::pl_sees( const tripoint &t, const int max_range ) const
           map_cache.sm[t.x][t.y] > 0.0 );
 }
 
+template<int xx, int xy, int xz, int yx, int yy, int yz, int zz,
+         float(*calc)(const float &, const float &, const int &),
+         bool(*check)(const float &, const float &)>
+void cast_zlight(
+    float (&output_cache)[MAPSIZE*SEEX][MAPSIZE*SEEY],
+    const std::array<const float (*)[MAPSIZE*SEEX][MAPSIZE*SEEY], OVERMAP_LAYERS> &input_arrays,
+    const tripoint &offset, const int offset_distance, const int target_z,
+    const std::function<bool(const tripoint &)> &floor_check,
+    const float numerator = 1.0f, const int row = 1,
+    float start_major = 1.0f, const float end_major = 0.0f,
+    float start_minor = 1.0f, const float end_minor = 0.0f,
+    double cumulative_transparency = LIGHT_TRANSPARENCY_OPEN_AIR )
+{
+    float radius = 60.0f - offset_distance;
+    if( start_major < end_major || start_minor < end_minor ) {
+        return;
+    }
+
+    /*
```

Comment:
I don't think we ever want something like this, we always want to determine what the player can see, whether we draw it or not.  I might be misunderstanding though.